api/v1alpha1/serviceintegration_types.go

```diff
@@ -25,26 +28,40 @@ type ServiceIntegrationSpec struct {
 	Project string `json:"project"`
 
 	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Value is immutable"
-	// +kubebuilder:validation:Enum=datadog;kafka_logs;kafka_connect;metrics;dashboard;rsyslog;read_replica;schema_registry_proxy;jolokia;internal_connectivity;external_google_cloud_logging;datasource;clickhouse_postgresql;clickhouse_kafka;logs;external_aws_cloudwatch_metrics
-	// Type of the service integration
+	// +kubebuilder:validation:Enum=alertmanager,autoscaler,caching,cassandra_cross_service_cluster,clickhouse_kafka,clickhouse_postgresql,dashboard,datadog,datasource,external_aws_cloudwatch_logs,external_aws_cloudwatch_metrics,external_elasticsearch_logs,external_google_cloud_logging,external_opensearch_logs,flink,flink_external_kafka,internal_connectivity,jolokia,kafka_connect,kafka_logs,kafka_mirrormaker,logs,m3aggregator,m3coordinator,metrics,opensearch_cross_cluster_replication,opensearch_cross_cluster_search,prometheus,read_replica,rsyslog,schema_registry_proxy,stresstester,thanosquery,thanosstore,vmalert
+	// Type of the service integration accepted by Aiven API. Some values may not be supported by the operator
```

Comment:
Please have a look at this: it is really hard to say which types can be supported without user configuration. I think it is better put the whole list and leave a warning here.